content/style-injector.js

```diff
@@ -94,40 +106,56 @@ function createStyleInjector({compare, setStyleContent, onUpdate}) {
     style.el.remove();
   }
 
-  function update({id, code}) {
+  function _update({id, code}) {
     const style = table.get(id);
     if (style.code === code) return;
     style.code = code;
     // workaround for Chrome devtools bug fixed in v65
-    // https://github.com/openstyles/stylus/commit/0fa391732ba8e35fa68f326a560fc04c04b8608b
-    let oldEl;
-    if (CHROME < 3321) {
-      oldEl = style.el;
+    if (isChromePre65) {
+      const oldEl = style.el;
       oldEl.id = '';
-      style.el = createStyle(id);
+      style.el = createStyle(id, code);
       oldEl.parentNode.insertBefore(style.el, oldEl.nextSibling);
-      style.el.disabled = !enabled;
+      oldEl.remove();
+    } else {
+      style.el.textContent = code;
+      style.el.sheet.disabled = !enabled;
+    }
+  }
+
+  function _supersede(domId) {
+    const el = document.getElementById(domId);
+    if (el) {
+      // remove if it looks like our style that wasn't cleaned up in orphanCheck
+      // (note, Firefox doesn't orphanize content scripts at all so orphanCheck will never run)
+      if (el.localName === 'style' && el.classList.contains('stylus')) {
+        el.remove();
+      } else {
+        el.id += ' superseded by Stylus';
+      }
     }
-    return setStyleContent(style.el, code, !enabled)
-      .then(() => oldEl && oldEl.remove());
   }
 
-  function createStyle(id) {
+  function createStyle(id, code) {
+    if (!creationDoc) _initCreationDoc();
     let el;
     if (document.documentElement instanceof SVGSVGElement) {
       // SVG document style
-      el = document.createElementNS('http://www.w3.org/2000/svg', 'style');
+      el = creationDoc.createElementNS('http://www.w3.org/2000/svg', 'style');
     } else if (document instanceof XMLDocument) {
       // XML document style
-      el = document.createElementNS('http://www.w3.org/1999/xhtml', 'style');
+      el = creationDoc.createElementNS('http://www.w3.org/1999/xhtml', 'style');
     } else {
       // HTML document style; also works on HTML-embedded SVG
-      el = document.createElement('style');
+      el = creationDoc.createElement('style');
     }
     el.id = `${PREFIX}${id}`;
     el.type = 'text/css';
     // SVG className is not a string, but an instance of SVGAnimatedString
     el.classList.add('stylus');
+    el.textContent = code;
+    el.disabled = !enabled; // https://github.com/openstyles/stylus/issues/693
```

Comment:
The disabled state has to be set after the element is attached to DOM so this won't work.

Reply:
Yep, my bad.